doc/dev/engsys.md

```diff
@@ -0,0 +1,73 @@
+# Engineering System Checks
+
+* [mypy](#mypy)
+* [black](#black)
+* [autorest automation](#autorest-automation)
```

Comment:
Should we also add opt-in to running samples in live CI?

Reply:
Added